### PR TITLE
Addresses various visual QA items from the collections work

### PR DESCRIPTION
**Pull request by sweir27**
This PR addresses a bunch of visual QA items from our last session!

The biggest change is the introduction of [`react-native-modal`](https://github.com/react-native-community/react-native-modal).

Reasoning:
- It appears well-maintained.
- It extends react native's default `Modal` implementation and has a super simple API. Should RN decide to add these features into their implementation, it should be easy enough to go back.
- It makes things like "have a background that fades in + a modal sheet that slides up" _incredibly_ easy (it's built in!). If we want to add more facilities, especially as we built out more interactivity in the app, this feels like a good addition.

Cons:
- It's _one more dep_. I was stuck on getting the behavior mentioned ☝️to feel seamless and it definitely felt cleaner/better to use this library vs. implementing the behavior on our own, but I'd love to get the input of MX folks (@ds300 , @ashfurrow , @pvinis , etc.)

(assigning myself just until we reach a decision on whether or not to include this new dependency!)

### CODE REVIEW

File: src/lib/Components/ArtworkFilterOptions/SingleSelectOption.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ export const SingleSelectOptionScreen: React.SFC<SingleSelectOptionScreenProps>
         </Sans>
         <Box></Box>
       </FilterHeader>
-      <Flex mb={120}>
+      <Flex mb="125px">
```
Comment (sweir27) [inline src/lib/Components/ArtworkFilterOptions/SingleSelectOption.tsx:46]: To account for extra space added below the `Apply` button.

Confirmation that it looks okay when you get to the bottom of the list:
image

File: src/lib/Components/FilterModal.tsx [hunk 2]
```diff
@@ -53,9 +54,9 @@ export const FilterModalNavigator: React.SFC<FilterModalProps> = props => {
   return (
     <>
       {isFilterArtworksModalVisible && (
-        <RNModal animationType="fade" transparent={true} visible={isFilterArtworksModalVisible}>
+        <Modal isVisible={isFilterArtworksModalVisible} style={{ margin: 0 }}>
```
Comment (sweir27) [inline src/lib/Components/FilterModal.tsx:57]: Addresses: https://artsyproduct.atlassian.net/browse/FX-1915 and https://artsyproduct.atlassian.net/browse/FX-1918

modal-animate

File: src/lib/Components/FilterModal.tsx [hunk 3]
```diff
@@ -266,20 +267,13 @@ export const OptionListItem = styled(Flex)`
...
-  border-top-right-radius: ${space(1)};
-`
-
 const ModalInnerView = styled.View`
   flex-direction: column;
   background-color: ${color("white100")};
   height: 75%;
   border-top-left-radius: ${space(1)};
   border-top-right-radius: ${space(1)};
+  overflow: hidden;
```
Comment (sweir27) [inline src/lib/Components/FilterModal.tsx:276]: Necessary for the border radius on the filter sheet.

File: src/lib/Scenes/Collection/Collection.tsx [hunk 4]
```diff
@@ -181,6 +181,7 @@ export const FilterArtworkButton = styled(Flex)<{ isFilterCountVisible: boolean
   align-items: center;
   justify-content: center;
   flex-direction: row;
+  box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.12);
```
Comment (sweir27) [inline src/lib/Scenes/Collection/Collection.tsx:184]: Addresses: https://artsyproduct.atlassian.net/browse/FX-1917

image

File: src/lib/Scenes/Collection/Components/FeaturedArtists.tsx [hunk 5]
```diff
@@ -73,7 +73,7 @@ export class FeaturedArtists extends React.Component<FeaturedArtistsProps, {}> {
         <Flex justifyContent="space-between" pb={15} flexDirection="row">
           <Sans size="4">{headlineLabel}</Sans>
           {artists.length > artistCount && (
-            <TouchableHighlight
+            <TouchableOpacity
```
Comment (sweir27) [inline src/lib/Scenes/Collection/Components/FeaturedArtists.tsx:76]: Addresses https://artsyproduct.atlassian.net/browse/FX-1920

view-all

File: src/lib/Scenes/Collection/Screens/CollectionHeader.tsx [hunk 6]
```diff
@@ -19,7 +19,7 @@ export const CollectionHeader: React.SFC<CollectionHeaderProps> = props => {
   const defaultHeaderUrl = image?.edges[0]?.node?.image?.url || ""
   const url = headerImage ? headerImage : defaultHeaderUrl
   const { width: screenWidth } = Dimensions.get("window")
-  const collectionTitleMargin = (collectionDescription || "").length < 1 ? 2 : 0
+  const collectionTitleMargin = (collectionDescription || "").length < 1 ? 2 : 1
```
Comment (sweir27) [inline src/lib/Scenes/Collection/Screens/CollectionHeader.tsx:22]: Addresses: https://artsyproduct.atlassian.net/browse/FX-1922

image

File: src/lib/Components/FilterModal.tsx [hunk 7]
```diff
@@ -289,6 +283,7 @@ export const ClearAllButton = styled(TouchableOpacity)``
 export const ApplyButton = styled(Button)``
 export const ApplyButtonContainer = styled(Box)`
   padding: 20px;
+  padding-bottom: 30px;
```
Comment (sweir27) [inline src/lib/Components/FilterModal.tsx:286]: Addresses: https://artsyproduct.atlassian.net/browse/FX-1913

image

**Comment by ashleyjelks** [comment]
These changes look great @sweir27!

**Comment by sweir27** [comment]
The RFC was approved: https://github.com/artsy/README/issues/307 !

Will merge in on 🍏 !

**Pull request merged**